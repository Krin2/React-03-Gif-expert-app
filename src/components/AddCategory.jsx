import { useState } from "react"

export const AddCategory = ({ setCategories }) => {

  // Hooks
  const [inputValue, setInputValue] = useState('')

  // Funciones
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  }

  const onSubmit = (event) => {
    event.preventDefault();

    // Verifico que haya contenido para agregar con mas de 1 caracter
    if ( inputValue.trim().length <=1) return;

    // agrego la categoria
    setCategories((categories) => [inputValue, ...categories]);
    
    // borro el input para escribir el siguiente valor
    setInputValue('');
  }
  return (
    <form onSubmit={ onSubmit }>
      <input
        type='text' 
        placeholder='Buscar Gif'
        value={ inputValue }
        onChange={ onInputChange }
      />
    </form>
  )
}
