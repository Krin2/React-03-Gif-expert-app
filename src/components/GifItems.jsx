import PropTypes from 'prop-types';

export const GifItems = ({ url, title}) => {
  return (
    <div className="card">
      <img src={ url } alt={ title } />
      <p>{ title }</p>
    </div>
  )
}

GifItems.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}