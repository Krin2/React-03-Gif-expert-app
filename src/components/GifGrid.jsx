/*
  Este componente contiene los Gif para una categoria seleccionada.
  El componente hace:
  - la busqueda de los gif
  - les coloca el titulo
  - los presenta
*/
import { GifItems } from "./GifItems";
import { useFetchGifs } from "../hooks/useFetchGifs";


export const GifGrid = ({ category }) => {

  // Hooks
  const { images, isLoading } = useFetchGifs(category);
  
  return (
    <>
      <h3>{ category }</h3>
      <div className={'card-grid'}>
        {
          images.map( (image) => (
            <GifItems 
              key={ image.id }
              {...image} />
          ))
        }
      </div>
    </>
  )
}
