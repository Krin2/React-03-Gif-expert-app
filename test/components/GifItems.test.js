import { render, screen } from "@testing-library/react";
import { GifItems } from "../../src/components/GifItems";

describe('Test del componente GifItems', () => {
  const title = 'mi titulo';
  const url = 'https://abc.com/de';

  test('Debe coincidir con el snapshot', () => {
    const { container } = render(<GifItems url= {url} title={ title }/>);

    expect(container).toMatchSnapshot();
  });

  test('Debe recibir la propiedad url en la imagen', () => {
    render(<GifItems url={ url } title={ title }/>);
    const {src, alt} = screen.getByRole("img");

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Debe recibir la propiedad title', () => {
    render(<GifItems url={ url } title={ title }/>);
    
    expect(screen.getAllByText(title)).toBeTruthy();
  });
})