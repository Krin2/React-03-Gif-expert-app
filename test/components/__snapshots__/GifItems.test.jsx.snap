// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test del componente GifItems Debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="mi titulo"
      src="https://abc.com/de"
    />
    <p>
      mi titulo
    </p>
  </div>
</div>
`;
